proprietary/design-tokens/build/build-tokens.mjs

```diff
@@ -0,0 +1,244 @@
+const getTokensByViewport = (groups, viewport) => filterTokenSets(groups['viewport'].find(({name}) => name.includes(viewport)).selectedTokenSets);
+
+/**
+ * Creates product combinations from groups of tokens
+ * Product tokens combined with Mode tokens and Viewport tokens
+ * @param {object} groups 
+ * @returns {Object.<string, Array.<string>>} Object with path as key and a array of combined tokensets
+ */
+const createProducts = (groups) => {
+  const viewports = groups['viewport'].map(({name}) => cleanName(name));
+  const modes = groups['mode'].map(({name}) => cleanName(name));
+  const combined = modes.map((mode) => [mode, [...viewports]]);
+  const mappings = groups['nl-mapping'].map(({name}) => name);
+
+  const logiusProducts = groups['product']
+    .filter(({name}) => cleanName(name) !== 'nldoc')
+    .flatMap(({name, selectedTokenSets}) => {
+      const productCombinations = getProductCombinations(name, combined);
+
+      selectedTokenSets = filterTokenSets(selectedTokenSets);
+      
+      return productCombinations.map(({name, mode, viewport}) => {
+        const modeTokens = getTokensByMode(groups, mode);
+        const viewportTokens = getTokensByViewport(groups, viewport);
+
+        return [`${name}/${mode}/${viewport}`, [...selectedTokenSets, ...modeTokens, ...viewportTokens]];
+      });
+    }
+  );
+
+  const nldocProducts = groups['nldoc']
+    .flatMap(({name, selectedTokenSets}) => {
+      const productCombinations = getProductCombinations(name, combined);
+
+      const nlDocTokenSets = filterTokenSets(groups['product'].find(({name}) => cleanName(name) === 'nldoc').selectedTokenSets);
+      selectedTokenSets = filterTokenSets(selectedTokenSets);
+
+      return productCombinations.map(({name, mode, viewport}) => {
+        const modeTokens = getTokensByMode(groups, mode);
+        const viewportTokens = getTokensByViewport(groups, viewport);
+
+        return [`nldoc/${name ? name + '/' : ''}${mode}/${viewport}`, [...nlDocTokenSets, ...selectedTokenSets, ...modeTokens, ...viewportTokens]];
+      });
+    }
+  );
+
+  return {
+    ...Object.fromEntries(logiusProducts), 
+    ...Object.fromEntries(nldocProducts),
+  };
+}
+
+/**
+ * Our version of permutateThemes 
+ * @param {object} themes parsed JSON object from $themes.json
+ * @returns Object with path as key and a array of combined tokensets
+ */
+const permutateThemes = (themes) => {
+  const themeGroups = getThemeGroupObjects(themes);
+
+  return createProducts(themeGroups);
+};
+
+/**
+ * Builds tokens
+ *
+ * @async
+ * @returns 
+ */
+export const buildTokens = async () => {
+  console.info('Building Tokens');
+
+  const $themes = JSON.parse(await readFile(`${IMPORTED_SRC_FOLDER}/$themes.json`, 'utf-8'));
+  const themes = permutateThemes($themes);
+
+  const configs = Object.entries(themes).map(([name, tokenSets]) => ({
+    source: [
+      // `${MANUAL_SRC_FOLDER}/missingTokens.json`,
```

Comment:
Ik zou deze er toch graag in houden. Zo ook de `light/dark` variant. Ook als ze leeg zijn. Op het moment dat we bezig zijn en we missen een paar tokens hoeven we niet op UX te wachten. Of als we willen experimenteren.